src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Emitter.cs

```diff
@@ -249,13 +249,18 @@ private void GenCases(LoggerMethod lm, string nestedIndentation)
                 foreach (LoggerParameter p in lm.TemplateParameters)
                 {
                     string name = p.Name;
```

Comment:
>string name = p.Name; [](http://example.com/codeflow?start=20&length=21)

can't we just make this `string name = p.CodeName` and avoid the next line check? I think when there is no `@` Name will be equal CodeName. right?

Reply:
Right, however if I change to it, this code fails:
``` c#
if (lm.TemplateMap.ContainsKey(name))
{
    // take the letter casing from the template
    name = lm.TemplateMap[name];
}
```

I'm changing this part to:

``` c#
string name = p.CodeName;

if (lm.TemplateMap.ContainsKey(p.Name))
{
    // take the letter casing from the template
    name = lm.TemplateMap[name];
}
```

This will mimic old behavior while maintaining support for '@' symbols

Reply:
Just update code, when you have time, please take a look.

Reply:
> however if I change to it, this code fails:

Why did it fail? I am seeing the code is initializing the `TemplateMap` from the template which should have CodeName. right? anyway, the current code is incorrect because you are checking `lm.TemplateMap.ContainsKey(p.Name)` and then using `name = lm.TemplateMap[name];` which can throw if `name` != `p.Name`.

Reply:
You're right.
Debugging the code (it is very hard to debug using only vscode btw), I got the following:
```c#

// CodeName = @myevent, p.Name = myevent, tm.TemplateMap[p.CodeName] = @myevent
// CodeName = otherevent, p.Name = otherevent, tm.TemplateMap[p.Name] = otherevent, tm.TemplateMap[p.CodeName] = otherevent
// CodeName = a1, p.Name = a1, tm.TemplateMap[p.Name] = A1, tm.TemplateMap[p.CodeName] = A1
```

So I think it is safe to use CodeName to find correct casing from template.